rosidl_generator_cpp/resource/msg__traits.hpp.em

```diff
@@ -61,6 +63,79 @@ for member in message.structure.members:
 namespace rosidl_generator_traits
 {
 
+void to_yaml(
```

Comment:
Should be made inline.

Reply:
I don't see why. The idea is that all the implementations should be moved into a `.cpp` file anyway.

Reply:
Inline works as a linkage specifier and must always be used in functions defined in header files to avoid violating the "one definition rule" (ODR).
Methods and friend functions defined in a header files are implicitly inline.
See [ODR](https://en.cppreference.com/w/cpp/language/definition#One_Definition_Rule), [inline](https://en.cppreference.com/w/cpp/language/inline).
e.g.:
```cpp
# file1.cpp
#include "package_name/msg/my_msg.hpp"
...
```
```cpp
# file2.cpp
#include "package_name/msg/my_msg.hpp"
...
```
```cpp
# CMakeLists.txt
...
add_library(<lib_name>
  <path_to_file1.cpp> <path_to_file2.cpp>)
...
```

will violate ODR and produce linkage errors (this ODR violation can also happen with linking with a dependency, using `inline` avoids the problem).

> The idea is that all the implementations should be moved into a .cpp file anyway.

Yeah, so if we want to do that we should do it before galactic, as removing the `inline` specifier will break ABI.

Reply:
> Inline works as a linkage specifier and must always be used in functions defined in header files to avoid violating the "one definition rule" (ODR).

Indeed :+1:

> Yeah, so if we want to do that we should do it before galactic, as removing the inline specifier will break ABI.

That was the idea. But if preferred we can first add this API with `inline` and then break ABI when moving it into `.cpp` files: done in 77da551af6ab2a9647e16d67c580f66ed9db1b1c.